tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -100,6 +109,50 @@ def wait(self):
         return False
 
 
+class WebDriverBidiEventsProtocolPart(BidiEventsProtocolPart):
+    _subscriptions = []
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.webdriver = None
+
+    def setup(self):
+        self.webdriver = self.parent.webdriver
+
+    async def subscribe(self, events, contexts):
+        self.logger.info("Subscribing to events %s in %s" % (events, contexts))
+        self._subscriptions.append((events, contexts))
```

Comment:
A problem with this being a list is that something like
```
subscribe(["event"], "context")
subscribe(["event"], "context")
unsubscribe_all()
```

is going to end up throwing an error during unsubscription because we'll try to remove a subscription that doesn't exist. Unfortunately I think you probably need to use the return value of the command to discover the new contexts for which the event was enabled, and store only those.

Reply:
Good point! I [updated](https://github.com/web-platform-tests/wpt/pull/45823/files#diff-e5a8911dd97e0352b1b26d8ce6ef0a92b25378f7c9e79371a1eb1b1834bc9a8dR163) the logic. I can see the following edge cases:
1. Subscription is done with wrong arguments. In this case the subscription will not happen at all, and nothing will be added to the `_subscriptions`.
2. Subscription is done for the sub-context which is removed later on. To handle this case, I add top-level contexts to the `_subscriptions`.